packages/sdk/src/adapters/multipeer/session.ts

```diff
@@ -241,14 +241,24 @@ export class Session extends EventEmitter {
 	}
 
 	public cacheInitializeActorMessage(message: InitializeActorMessage) {
-		if (!this.actorSet[message.payload.actor.id]) {
+		const syncActor = this.actorSet[message.payload.actor.id];
+		if (!syncActor) {
 			const parent = this.actorSet[message.payload.actor.parentId];
 			this.actorSet[message.payload.actor.id] = {
 				actorId: message.payload.actor.id,
 				exclusiveToUser: parent && parent.exclusiveToUser
 					|| message.payload.actor.exclusiveToUser,
 				initialization: deepmerge({ message }, {})
 			};
+		// update reserved actor init message with something the client can use
+		} else if (syncActor.initialization.message.payload.type === 'x-reserve-actor') {
+			// send real init message, but with session's initial actor state
+			message.payload = {

```

Comment:
>message [](start = 3, length = 7)

Safer to make a deepmerge copy of message here before modifying it.

Reply:
I'm not sure it does in context. The actor patch in the create message is a strictly older version of the one in the session, I don't think we want anything from it.

---
In reply to: [316825178](https://github.com/microsoft/mixed-reality-extension-sdk/pull/423#discussion_r316825178) [](ancestors = 316825178)

Reply:
I vaguely recall that making a deep copy of the entire message was important to do here, but don't remember the reason now. We should preserve that behavior.